mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -86,8 +86,8 @@ public partial class Socket : IDisposable
 		 */
 		internal EndPoint seed_endpoint = null;
 
-		internal Queue<SocketAsyncWorker> readQ = new Queue<SocketAsyncWorker> (2);
-		internal Queue<SocketAsyncWorker> writeQ = new Queue<SocketAsyncWorker> (2);
+		internal Queue<KeyValuePair<IntPtr, IOSelectorJob>> readQ = new Queue<KeyValuePair<IntPtr, IOSelectorJob>> (2);
```

Comment:
It won't but it make it easier to understand where the handle comes from in https://github.com/mono/mono/pull/2060/files#diff-df7cce0e9bc50b01c49c33e08019b5b5R145